dc/s2n-quic-dc/src/stream/server/tokio/uds.rs

```diff
@@ -0,0 +1,207 @@
+        let meta = event::api::ConnectionMeta {
+            id: 0, // TODO use an actual connection ID
+            timestamp: now.into_event(),
+        };
+        let info = event::api::ConnectionInfo {};
+        let subscriber_ctx = self
+            .env
+            .subscriber()
+            .create_connection_context(&meta, &info);
+
+        let recv_buffer = recv::buffer::Local::new(buffer, None);
+        let recv_buffer: either::Either<_, Channel> = Either::A(recv_buffer);
+        let secret_control = vec![]; // this is only used while returning an error from endpoint::accept_stream
+
+        let export_secret: ExportSecret =
+            decoded_packet.export_secret().try_into().map_err(|e| {
+                std::io::Error::new(
+                    ErrorKind::InvalidInput,
+                    format!("Error parsing export secret {:?}", e),
+                )
+            })?;
+
+        // create app pair
+        let key_id = initial_packet.credentials.key_id;
+        let initiator = Initiator::Remote;
+        let secret = Secret::new(
+            decoded_packet.ciphersuite(),
+            decoded_packet.version_tag().into(),
+            Type::Server,
+            &export_secret,
+        );
+        let application = ApplicationPair::new(
+            &secret,
+            key_id,
+            initiator,
+            // Dedup should be done in manager
+            Dedup::disabled(),
+        );
+
+        let control = None;
+        let crypto = Bidirectional {
+            credentials: initial_packet.credentials,
+            application,
+            control,
+        };
+
+        let local_port = tcp_stream.local_addr()?.port();
+        let socket = LazyBoundStream::Tokio(tcp_stream);
+        let peer = env::tcp::Reregistered {
+            socket,
+            peer_addr: remote_address.into(),
+            local_port,
+            recv_buffer,
+        };
+        let stream_builder = match endpoint::accept_stream(
+            now,
+            &self.env,
+            peer,
+            &initial_packet,
+            &self.map,
+            subscriber_ctx,
+            None,
+            crypto,
+            decoded_packet.application_params().clone(),
+            secret_control,
+        ) {
+            Ok(stream) => stream,
+            Err(error) => {
+                return Err(std::io::Error::new(
+                    ErrorKind::InvalidData,
+                    format!("Failed to accept stream, err: {:?}", error.error),
+                ));
+            }
+        };
+
+        {
+            let remote_address: SocketAddress = stream_builder.shared.remote_addr();
+            let remote_address = &remote_address;
+            let creds = stream_builder.shared.credentials();
+            let credential_id = &*creds.id;
+            let stream_id = creds.key_id.as_u64();
+            publisher.on_acceptor_tcp_stream_enqueued(event::builder::AcceptorTcpStreamEnqueued {
```

Comment:
I think we'll want a different event on the application side... maybe *de*queued?